### PR TITLE
Feat/migrate react hooks

### PR DESCRIPTION
**Pull request by AugustoCalaca**
I started the migration to hooks api

* update react and react-dom to v16.13.1
* add eslint plugin enforces the rules of hooks
* refactor the fist file Agenda.js to hooks

The main idea of ​​this PR is to be the starting point for migrate the entire repo and finally close this issue #1620

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,6 +2,9 @@ parser: babel-eslint
 extends:
   - jason/react
   - prettier
+  - plugin:react-hooks/recommended
```
Comment (jquense) [inline .eslintrc:5]: you've got this over specified here. If we add the rules manually like you have below the extends isn't needed here

File: src/Agenda.js [hunk 2]
```diff
@@ -10,75 +10,25 @@ import { navigate } from './utils/constants'
...
-  renderDay = (day, events, dayKey) => {
+  const renderDay = (day, events, dayKey) => {
     let {
       selected,
       getters,
       accessors,
       localizer,
       components: { event: Event, date: AgendaDate },
-    } = this.props
+    } = props
```
Comment (jquense) [inline src/Agenda.js:31]: props should be spread out at the top level (the component function body), so each function doesn't have to respread out the ones they need

Comment (AugustoCalaca) [reply src/Agenda.js]: great

**Comment by hiroki-shiotsuka** [comment]
Still, neither `jquense/` nor `/intljusticemission` repo hadn't removed UNSAFE_componentWillReceiveProps() function yet. It won't work on strict mode.

**Pull request merged**